dali/operators/generic/slice/slice_attr.h

```diff
@@ -103,16 +103,26 @@ class SliceAttr {
         for (size_t i = 0; i < axes.size(); i++) {
           auto dim = axes[i];
           float anchor_val = slice_anchor_data[i];
-          if (normalized_anchor_)
-            anchor_val *= shape[dim];
           float shape_val = slice_shape_data[i];
-          if (normalized_shape_)
+          int64_t slice_end;
+          // special case - minimize the floating point error by multiplying only once after sum
+          if (normalized_anchor_ && normalized_shape_) {
+            slice_end = std::llround((anchor_val + shape_val) * shape[dim]);
+            anchor_val *= shape[dim];
             shape_val *= shape[dim];
-          int64_t slice_end = static_cast<int64_t>(anchor_val + shape_val);
+          } else {
+            if (normalized_anchor_) {
+              anchor_val *= shape[dim];
+            }
+            if (normalized_shape_) {
+              shape_val *= shape[dim];
+            }
+            slice_end = std::llround(anchor_val + shape_val);
```

Comment:
This is not correct, especially for really big data. Cast to double or calculate slice_end using integer start/end in respective `if` branches.

Reply:
Done